src/stubs/sm_entry.s

```diff
@@ -7,35 +7,42 @@
     ; r7: return address
 __sm_entry:
     ; === need a secure stack to handle IRQs ===
-    dint
+    ; dint
+    ; First remove ssa base addr to notify violation handler to not use this SMs violation pointer
+    mov #0, &__sm_ssa_base_addr
+    ; back up r15
+    mov r15, &__sm_tmp
+    ; set up clix length and call clix (word 0x1389)
+    mov #100, r15 
+    .word 0x1389
```

Comment:
not sure anymore: how will `clix` behave when ATOMICITY_MONITOR=0 in sancus-core? It should be a NOP right?

Reply:
also a bit tricky to have 100 hard-coded here maybe? let's at least indicate with a big line in comments from where to where it runs atomically?

Reply:
Yes
- `ATOMICITY_MONITOR=0` : `clix` is a `NOP and `dint` disables interrupts
- `ATOMICITY_MONITOR=1` : `clix` disables interrupts for X cylces and `dint` is a `NOP`

Reply:
Done in latest commit